pandas/io/parsers.py

```diff
@@ -1455,8 +1456,8 @@ def _get_name(icol):
                 return icol
 
             if col_names is None:
-                raise ValueError(('Must supply column order to use %s as '
-                                  'index') % str(icol))
+                raise ValueError(('Must supply column order to use {icol} as '
+                                  'index').format(icol=str(icol)))
```

Comment:
FYI you can do something like `'... {icol!s} ...'.format(icol=icol)` instead of explicitly casting to string.  See the "conversion flags" examples towards the end of the [Format String Syntax](https://docs.python.org/3.6/library/string.html#format-string-syntax) section of the docs for more info.